server/node/src/main/java/org/kaaproject/kaa/server/admin/servlet/ConfigurationDownloadServlet.java

```diff
@@ -0,0 +1,63 @@
+/**
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.kaaproject.kaa.server.admin.servlet;
+
+import com.google.gwt.thirdparty.json.JSONArray;
+import com.google.gwt.thirdparty.json.JSONException;
+import com.google.gwt.thirdparty.json.JSONObject;
+import org.kaaproject.kaa.common.dto.ConfigurationRecordDto;
+import org.kaaproject.kaa.server.admin.shared.servlet.ServletParams;
+import org.kaaproject.kaa.server.common.dao.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.URLDecoder;
+
+public class ConfigurationDownloadServlet extends HttpServlet implements Servlet, ServletParams {
```

Comment:
It is easier to add this logic to https://github.com/kaaproject/kaa/blob/develop/server/node/src/main/java/org/kaaproject/kaa/server/admin/servlet/RecordServlet.java
This will help to avoid code duplications